extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js

```diff
@@ -44,14 +45,84 @@ require([
 		});
 	}
 
+	function processData(threads, baseUrl, upvoteUrl) {
+		var ret = [],
+			i,
+			thread,
+			userData;
+
+		for (i in threads) {
+			thread = threads[i];
+			userData = thread._embedded.userData[0];
+
+			ret.push({
+				author: thread.createdBy.name,
+				authorAvatar: thread.createdBy.avatarUrl,
+				commentCount: thread.postCount,
+				content: thread.rawContent,
+				createdAt: thread.creationDate.epochSecond, // TODO: format to string
+				forumName: $.msg( 'embeddable-discussions-forum-name', thread.forumName),
+				id: thread.id,
+				firstPostId: thread.firstPostId,
+				index: i,
+				link: '/d/p/' + thread.id,
+				shareUrl: baseUrl + 'd/p/' + thread.id,
+				upvoteUrl: upvoteUrl + thread.firstPostId,
+				title: thread.title,
+				upvoteCount: thread.upvoteCount,
+				hasUpvoted: userData.hasUpvoted,
+			});
+		}
+
+		return ret;
+	}
+
+	function performRequest($elem) {
+		var requestUrl = $elem.attr('data-requestUrl'),
+			requestData = JSON.parse($elem.attr('data-requestData'));
+
+		$.ajax({
+			type: 'GET',
+			url: requestUrl,
+			xhrFields: {
+				withCredentials: true
+			},
+		}).done(function (data) {
+			var threads = processData(data._embedded.threads, requestData.baseUrl, requestData.upvoteRequestUrl);
+
+			$elem.html(mustache.render(templates.DiscussionThreads, {
+				threads: threads,
+				columnsDetailsClass: requestData.columnsDetailsClass,
+				replyText: $.msg('embeddable-discussions-reply'),
+				shareText: $.msg('embeddable-discussions-share'),
+				showAll: $.msg('embeddable-discussions-show-all'),
+				upvoteText: $.msg('embeddable-discussions-upvote'),
+				zeroText: $.msg('embeddable-discussions-zero'),
+				zeroTextDetail: $.msg('embeddable-discussions-zero-detail'),
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Remove this trailing comma. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATrailingComma)